core/src/term/mod.rs

```diff
@@ -207,7 +207,7 @@ pub enum Term {
 
     /// An unresolved import.
     #[serde(skip)]
-    Import(OsString),
+    Import{path: OsString, typ: Option<LocIdent>},
```

Comment:
I'd suggest having `typ: InputFormat` here, and handling the supported types (including the auto-detection from the filename) while constructing the `Term`.

Reply:
1. How would it interact with optionally present format like `Nix`? Ideally it should recognize it and preserve during round-tripping and only fail when actually trying to import the file.
2. Does it move complexity to the `grammar.lalrpop`? (It's the first time I edit such a file)

Reply:
1. Good question. I think since `InputFormat::Nix` is experimental, it's ok to error if it isn't supported.
2. I'd suggest adding the actual logic in `core/src/parser/utils.rs`, so that the changes to `grammar.lalrpop` will be very simple.